strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/io/ArtifactOutputStream.java

```diff
@@ -15,18 +20,118 @@
 public class ArtifactOutputStream
         extends MultipleDigestOutputStream
 {
-
+    private Function<byte[], String> digestStringifier = MessageDigestUtils::convertToHexadecimalString;
     private ArtifactCoordinates coordinates;
+    /**
+     * Used to cache Artifact contents if needed.
+     */
+    private OutputStream cacheOutputStream;
+    private Function<OutputStreamFunction, ?> cahceOutputStreamTemplate = this::doWithOutputStream;
 
-    public ArtifactOutputStream(OutputStream source, ArtifactCoordinates coordinates) throws NoSuchAlgorithmException
+    public ArtifactOutputStream(OutputStream source,
+                                ArtifactCoordinates coordinates)
+        throws NoSuchAlgorithmException
     {
-        super(source);
+        super(source, new String[] {});
         this.coordinates = coordinates;
     }
 
+    public void setCahceOutputStreamTemplate(Function<OutputStreamFunction, ?> chahceOutputStreamTemplate)
```

Comment:
There's a typo here (and in a few other places -- please search for `cahce` to the the other places and correct them). The name of the method should be `setCacheOutputStreamTemplate`; the parameter's name is also misspelled.